### PR TITLE
Extract property names, validate expressions.

### PR DESCRIPTION
**Pull request by rictic**
- [x] CHANGELOG.md not updated, small improvement on top of #470

### CODE REVIEW

File: src/polymer/expression-scanner.ts [hunk 1]
```diff
@@ -128,6 +138,67 @@ export class DatabindingExpression {
...
   }
+
+  private _extract() {
+    if (this._expressionAst.body.length !== 1) {
+      this.warnings.push(this._validationWarning(
+          'Expected exactly one expression', this._expressionAst));
+      return;
+    }
+    const expressionStatement = this._expressionAst.body[0]!;
+    if (expressionStatement.type !== 'ExpressionStatement') {
```
Comment (justinfagnani) [inline src/polymer/expression-scanner.ts]: we know that we're always getting a statement as opposed to an expression?

File: src/polymer/expression-scanner.ts [hunk 2]
```diff
@@ -128,6 +138,67 @@ export class DatabindingExpression {
...
+      this.warnings.push(this._validationWarning(
+          'ExpressionStatement expected', expressionStatement));
```
Comment (justinfagnani) [inline src/polymer/expression-scanner.ts]: I don't think a user will understand a `ExpressionStatement`

File: src/polymer/expression-scanner.ts [hunk 3]
```diff
@@ -125,6 +135,67 @@ export class DatabindingExpression {
...
+    }
+    if (expression.type === 'CallExpression') {
+      this._handleSubExpression(expression.callee);
+      for (const arg of expression.arguments) {
+        this._handleSubExpression(arg);
+      }
+      return;
+    }
+    this.warnings.push(this._validationWarning(
+        `Unexpected node type: ${expression.type
```
Comment (justinfagnani) [inline src/polymer/expression-scanner.ts]: I don't think ast node types will be useful in warnings.

File: src/polymer/expression-scanner.ts (hunk 1 above)
Comment (rictic) [inline src/polymer/expression-scanner.ts]: Yep! Program#body is of type `Array<Statement | ModuleDeclaration>`

File: src/polymer/expression-scanner.ts [hunk 4]
```diff
@@ -128,6 +138,67 @@ export class DatabindingExpression {
...
+        this.warnings.push(this._validationWarning(
+            'Only the not (!) operator is supported.', expression));
+        return;
+      }
+      expression = expression.argument;
+    }
+    this._handleSubExpression(expression);
+  }
+
+  private _handleSubExpression(expression: estree.Node) {
```
Comment (rictic) [inline src/polymer/expression-scanner.ts]: Fair, named more consistently. Also added a flag to the method to track whether call expressions are allowed.

File: src/polymer/expression-scanner.ts (hunk 2 above)
Comment (rictic) [inline src/polymer/expression-scanner.ts]: Fair. I changed to a two sentence warning style in a couple places here. The first gives the problem in plain language, the second gives more technical detail that will be helpful to experts (including us, when people report issues).

File: src/polymer/expression-scanner.ts [hunk 5]
```diff
@@ -128,6 +138,67 @@ export class DatabindingExpression {
     return correctSourceRange(
         databindingRelativeSourceRange, this.locationOffset);
   }
+
+  private _extract() {
```
Comment (rictic) [inline src/polymer/expression-scanner.ts]: Done

File: src/polymer/expression-scanner.ts (hunk 3 above)
Comment (rictic) [inline src/polymer/expression-scanner.ts]: Changed to a two sentence pattern. Simple first sentence, more technical second sentence.

**Pull request merged**